packages/edit-site/src/components/sidebar-edit-mode/global-styles-sidebar.js

```diff
@@ -18,35 +17,25 @@ import { GlobalStylesMenuSlot } from '../global-styles/ui';
 import { unlock } from '../../private-apis';
 
 export default function GlobalStylesSidebar() {
-	const { shouldClearCanvasContainerView, isStyleBookOpened } = useSelect(
-		( select ) => {
-			const { getActiveComplementaryArea } = select( interfaceStore );
-			const _isVisualEditorMode =
-				'visual' === select( editSiteStore ).getEditorMode();
-
-			return {
-				isStyleBookOpened:
-					'style-book' ===
-					unlock(
-						select( editSiteStore )
-					).getEditorCanvasContainerView(),
-				shouldClearCanvasContainerView:
-					'edit-site/global-styles' !==
-						getActiveComplementaryArea( 'core/edit-site' ) ||
-					! _isVisualEditorMode,
-			};
-		},
-		[]
-	);
-
+	const { editorMode, editorCanvasView } = useSelect( ( select ) => {
```

Comment:
This change reinstates the file before https://github.com/WordPress/gutenberg/pull/50081 was merged given that this PR uses @noisysocks's original method of checking for the existence of the slot fill to determine whether to display the header.